File: app/src/main/java/com/arukai/uajpspeak/adapter/NavigationDrawerAdapter.java
package com.arukai.uajpspeak.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import androidx.recyclerview.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import com.arukai.uajpspeak.R;
import com.arukai.uajpspeak.model.NavDrawerItem;

public class NavigationDrawerAdapter extends RecyclerView.Adapter<NavigationDrawerAdapter.MyViewHolder> {
    List<NavDrawerItem> data;
    private LayoutInflater inflater;

    public NavigationDrawerAdapter(Context context, List<NavDrawerItem> data) {
        inflater = LayoutInflater.from(context);
        this.data = data;
    }

    public void delete(int position) {
        data.remove(position);
        notifyItemRemoved(position);
    }

    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.nav_drawer_row, parent, false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        NavDrawerItem current = data.get(position);
        holder.title.setText(current.getTitle());
        if(current.getSelected()){
            holder.title.setBackgroundResource(R.color.selectedCategory);
        } else {
            holder.title.setBackgroundResource(R.color.unselectedCategory);
        }
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    class MyViewHolder extends RecyclerView.ViewHolder {
        TextView title;

        public MyViewHolder(View itemView) {
            super(itemView);
            title = itemView.findViewById(R.id.title);
        }
    }
}

File: app/src/main/java/com/arukai/uajpspeak/adapter/NavigationDrawerAdapter.kt
package com.arukai.uajpspeak.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.arukai.uajpspeak.R
import com.arukai.uajpspeak.model.NavDrawerItem

class NavigationDrawerAdapter(
    context: Context,
    private val data: MutableList<NavDrawerItem>
) : RecyclerView.Adapter<NavigationDrawerAdapter.MyViewHolder>() {

    private val inflater: LayoutInflater = LayoutInflater.from(context)

    fun delete(position: Int) {
        data.removeAt(position)
        notifyItemRemoved(position)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view = inflater.inflate(R.layout.nav_drawer_row, parent, false)
        return MyViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val current = data[position]
        holder.title.text = current.title
        if (current.isSelected) {
            holder.title.setBackgroundResource(R.drawable.drawer_item_selected)
            holder.title.setTextColor(holder.itemView.context.getColor(R.color.selectedCategoryText))
            holder.title.setTypeface(null, android.graphics.Typeface.BOLD)
        } else {
            holder.title.setBackgroundResource(R.drawable.drawer_item_unselected)
            holder.title.setTextColor(holder.itemView.context.getColor(R.color.unselectedCategoryText))
            holder.title.setTypeface(null, android.graphics.Typeface.NORMAL)
        }
    }

    override fun getItemCount(): Int = data.size

    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val title: TextView = itemView.findViewById(R.id.title)
    }
}

